src/sources/datadog_agent/metrics.rs

```diff
@@ -228,6 +232,28 @@ fn decode_datadog_series_v2(
     Ok(metrics)
 }
 
+/// Builds Vector's `EventMetadata` from the received metadata. Currently this is only
+/// utilized for passing through origin metadata set by the Agent.
+fn get_event_metadata(metadata: Option<&Metadata>) -> EventMetadata {
+    metadata.map_or(EventMetadata::default(), |metadata| {
```

Comment:
Small nit - here and below, we should use `map_or_else` given that `EventMetadata::default()` is a function and `map_or` eagerly evaluates the default value.

```suggestion
    metadata.map_or_else(|| EventMetadata::default(), |metadata| {
```

Reply:
Also, we can reduce this to a single `map_or_else` to improve the readability of this code block. The two `map_or` calls tripped me up.

```
    metadata
        .map(|metadata| metadata.origin.as_ref())
        .flatten()
        .map_or_else(
            || EventMetadata::default(),
            |origin| {
                debug!(
                    "Deserialized origin_product: `{}` origin_category: `{}` origin_service: `{}`.",
                    origin.origin_product, origin.origin_category, origin.origin_service
                );
                EventMetadata::default().with_origin_metadata(
                    DatadogMetricOriginMetadata::default()
                        .with_product(origin.origin_product)
                        .with_category(origin.origin_category)
                        .with_service(origin.origin_service),
                )
            },
        )
```